src/transformers/models/mamba2/modeling_mamba2.py

```diff
@@ -51,6 +54,12 @@
 else:
     causal_conv1d_update, causal_conv1d_fn = None, None
 
+if is_flash_attn_2_available:
+    from flash_attn import flash_attn_with_kvcache
+    from flash_attn.layers.rotary import RotaryEmbedding
+else:
+    flash_attn_with_kvcache, RotaryEmbedding = None, None
```

Comment:
we just stick with the mamba SSM repo and use the rotary embedding implementation from FA

Reply:
not doing. abandoning `modeling_mamba2`